src/actions/result.js

```diff
@@ -53,17 +54,26 @@ export const getResult = ({ id }) => (dispatch) => {
   }
 };
 
-export const deleteResult = ({ id }) => (dispatch) => {
+export const deleteResult = ({ id, name }) => (dispatch) => {
   const flagDeletingResult = createFlag(FLAG_DELETING_RESULT);
   try {
-    dispatch(flagDeletingResult(true));
-
-    window.ipcRenderer.send(DELETE_RESULT_CHANNEL, { id });
-    window.ipcRenderer.once(DELETE_RESULT_CHANNEL, async (event, response) => {
-      dispatch(response);
-      dispatch(flagDeletingResult(false));
-      return getResults()(dispatch);
-    });
+    window.ipcRenderer.send(SHOW_CONFIRM_DELETE_PROMPT_CHANNEL, { name });
+    window.ipcRenderer.once(
+      SHOW_CONFIRM_DELETE_PROMPT_CHANNEL,
+      (e, shouldDelete) => {
+        if (shouldDelete) {
+          dispatch(flagDeletingResult(true));
+          window.ipcRenderer.send(DELETE_RESULT_CHANNEL, { id });
+          window.ipcRenderer.once(
+            DELETE_RESULT_CHANNEL,
+            async (event, response) => {
+              dispatch(response);
+            },
+          );
+          dispatch(flagDeletingResult(false));
```

Comment:
Move `dispatch(flagDeletingResult(false));` at the end of the `window.ipcRenderer.once` callback